jupyter_server/auth/identity.py

```diff
@@ -0,0 +1,155 @@
+"""Identity Provider interface
+
+This defines the _authentication_ layer of Jupyter Server,
+to be used in combination with Authorizer for _authorization_.
+
+.. versionadded:: 2.0
+"""
+import sys
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from tornado.web import RequestHandler
+from traitlets.config import LoggingConfigurable
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    try:
+        from typing_extensions import TypedDict
+    except ImportError:
+        TypedDict = Dict
+
+
+class IdentityModel(TypedDict):
+    # see the JupyterLab IUser model for definitions
```

Comment:
Do we want this to be the authoritative definition in the long run?

Reply:
Switched to User dataclass